skiko/src/awtMain/kotlin/org/jetbrains/skiko/SkiaLayer.awt.kt

```diff
@@ -109,11 +108,25 @@ actual open class SkiaLayer internal constructor(
         }
         @Suppress("LeakingThis")
         add(backedLayer)
+
         backedLayer.addHierarchyListener {
             if (it.changeFlags and HierarchyEvent.SHOWING_CHANGED.toLong() != 0L) {
                 checkShowing()
             }
         }
+
+        addPropertyChangeListener("graphicsContextScaleTransform") {
+            redrawer?.syncSize()
+            notifyChange(PropertyKind.ContentScale)
+
+            // Workaround for JBR-5259
+            if (hostOs == OS.Windows) {
+                SwingUtilities.invokeLater {
```

Comment:
I don't have this issue on JBR 17.0.4 - Windows 11, 2 monitors 1.5 + 1.0 scales. Tried with the reproducer in the issue.

Could you check, if we have a bug on JDK as well?

Reply:
Will do tomorrow!

Reply:
> One thing that always works though is to change the scale on a single monitor 

Thanks! I reproduced it on JBR/JRE with the SkiaAwtSample.

But can't reproduce it in a Compose application. Investigating...

If we will keep this hack, we need to implement it properly - cancelling a scheduled resize (to avoid hypothetical memory leaks, crashes, and concurrent changes):
```
private val fixContentJob: Job?

fixContentJob?.cancel()
fixContentJob = GlobalScope.launch(MainUIDispatcher) {
   ...
}

fun dispose() {
   ...
   fixContentJob?.cancel()
}
```

P.S. ignore CI crash on iOS

Reply:
> But can't reproduce it in a Compose application. Investigating...

I was wrongly reproduced it. Now it reproduces everywhere in Compose.

It seems like a regression after some Windows update. It worked in Windows 10 and may be in some Windows 11.

It seems, the hack doesn't work when we change the scale from 200 to 125, and almost works when we change from 125 to 200 (we have 10-20px stripes)

Reply:
I can in fact even reproduce the issue with any scale different from 100% in my VM. It looks to me like there is a third bug in the JBR which for some reason causes the backing buffer or the peer to be reallocated at 100% scale always when the scaling factor changes. Note that this is definitely different behavior from when the window is dragged between monitors, and I would consider this case a lot less frequent. As always, resizing the window helps, but this is not OK.

I have pushed the fix for the job scheduling, and I will continue the investigation about the 1x buffer tomorrow, when I have a proper Windows machine at hand which can compile custom JBRs etc.

Reply:
I believe I found a way to work around the remaining issue. I filed an additional [JBR ticket](https://youtrack.jetbrains.com/issue/JBR-5305/Canvas-is-placed-in-the-wrong-location-and-cut-off-and-no-property-change-is-fired-after-changing-the-monitor-scale-on-Windows) to track it. It's similar but not the same as the one with two monitors.

I ended up bringing back very similar `checkContentScale` logic to the one that we had before, but this time it's marked with the relevant ticket numbers so we can remove the workarounds once it's fixed at the root. Even with the workarounds in place, proper fixes at the root would be very desirable to avoid flickers and intermediate invalid states.

@igordmn let me know if this works for you and if we can mere it then!

Reply:
Thanks! I will look at it tomorrow.

> but this time it's marked with the relevant ticket numbers so we can remove the workarounds once it's fixed at the root

We can do that only if it is fixed in OpenJDK. Skiko (and Compose) supports not only JBR.

Reply:
I still have this issue (glitches after we change the scale)

![image](https://user-images.githubusercontent.com/5963351/219646898-29fea379-d3ad-4f80-82f1-bf4c6b077b21.png)

JDK 15, JBR 17

Does it work on your machine with the Clocks example?

Reply:
I just tested it with both JBR 17.0.4.1 and corretto 11.0.18, and it worked with both. Are you sure you were using JBR 17? The screenshot lists a version of 11.0.11+9-b1475.1, which seems to be from May 21? https://github.com/JetBrains/JetBrainsRuntime/tree/jb11_0_11-b1475

Reply:
On a new Windows configuration I can't reproduce it anymore. Maybe it is not always reproducible, maybe I was testing it wrongly.